ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -381,6 +382,55 @@ public PartitionId getRandomWritablePartition(String partitionClass, List<Partit
     return partitionSelectionHelper.getPartitions(partitionClass);
   }
 
+  @Override
+  public ReplicaId getNewReplica(String partitionIdStr, DataNodeId dataNodeId) {
+    ReplicaId newReplica = null;
+    logger.info("Getting ReplicaAddition ZNRecord from HelixPropertyStore in local DC.");
+    ZNRecord zNRecord = helixPropertyStoreInLocalDc.get(REPLICA_ADDITION_ZNODE_PATH, null, AccessOption.PERSISTENT);
+    if (zNRecord != null) {
+      String instanceName = getInstanceName(dataNodeId.getHostname(), dataNodeId.getPort());
+      Map<String, Map<String, String>> partitionToReplicas = zNRecord.getMapFields();
+      Map<String, String> replicaInfos = partitionToReplicas.get(partitionIdStr);
+      if (replicaInfos != null && replicaInfos.containsKey(instanceName)) {
+        long replicaCapacity = Long.valueOf(replicaInfos.get(REPLICAS_CAPACITY_STR));
+        String partitionClass = replicaInfos.get(PARTITION_CLASS_STR);
+        AmbryPartition mappedPartition = partitionNameToAmbryPartition.get(partitionIdStr);
+        if (mappedPartition == null) {
+          logger.info("Partition {} is currently not present in cluster map, creating a new partition.",
+              partitionIdStr);
+          mappedPartition =
+              new AmbryPartition(Long.valueOf(partitionIdStr), partitionClass, helixClusterManagerCallback);
```

Comment:
intellij suggests using `Long.parseLong` here to avoid unnecessary boxing

Reply:
fixed